prometheus_fastapi_instrumentator/instrumentation.py

```diff
@@ -119,78 +119,20 @@ def instrument(self, app: FastAPI):
-                labelnames=labels,
-                multiprocess_mode="livesum",
-            )
-
-        @app.middleware("http")
-        async def dispatch_middleware(request: Request, call_next) -> Response:
-            start_time = default_timer()
-
-            handler, is_templated = self._get_handler(request)
-            is_excluded = self._is_handler_excluded(handler, is_templated)
-            handler = (
-                "none" if not is_templated and self.should_group_untemplated else handler
-            )
-
-            if not is_excluded and self.should_instrument_requests_inprogress:
-                inprogress: Gauge
-                if self.inprogress_labels:
-                    inprogress = self.inprogress.labels(request.method, handler)
-                else:
-                    inprogress = self.inprogress
-                inprogress.inc()
-
-            response = None
-            status = "500"
-
-            try:
-                response = await call_next(request)
-                status = str(response.status_code)
-            except Exception as e:
-                raise e from None
-            finally:
-                if not is_excluded:
-                    duration = max(default_timer() - start_time, 0)
-
-                    if self.should_instrument_requests_inprogress:
-                        inprogress.dec()  # type: ignore
-
-                    if self.should_round_latency_decimals:
-                        duration = round(duration, self.round_latency_decimals)
-
-                    if self.should_group_status_codes:
-                        status = status[0] + "xx"
-
-                    info = metrics.Info(
-                        request=request,
-                        response=response,
-                        method=request.method,
-                        modified_handler=handler,
-                        modified_status=status,
-                        modified_duration=duration,
-                    )
-
-                    for instrumentation in self.instrumentations:
-                        instrumentation(info)
-
-            return response
-
+        app.add_middleware(
+            PrometheusInstrumentatorMiddleware,
+            should_group_status_codes=self.should_group_status_codes,
+            should_ignore_untemplated=self.should_ignore_untemplated,
+            should_group_untemplated=self.should_group_untemplated,
+            should_round_latency_decimals=self.should_round_latency_decimals,
+            should_respect_env_var=self.should_respect_env_var,
+            should_instrument_requests_inprogress=self.should_instrument_requests_inprogress,
+            round_latency_decimals=self.round_latency_decimals,
+            env_var_name=self.env_var_name,
+            inprogress_name=self.inprogress_name,
+            inprogress_labels=self.inprogress_labels,
+            instrumentations=self.instrumentations,
```

Comment:
```suggestion
            instrumentations=self.instrumentations,
            excluded_handlers=self.excluded_handlers,
```

This fixes the tests it seems